### ISSUE TITLE
Extract Variable insert position doesn't right

### ISSUE THREAD

**Issue by tjx666** [nicoespeon/abracadabra#819]
https://user-images.githubusercontent.com/41773861/219843731-46ede5e0-5b51-4fdb-8c43-b0ef85eed077.mov

before:

```typescript
const a = 1;
console.log(a);
console.log(a + 1);
console.log(a + 1);
```

result:

```typescript
const b = a + 1;
const a = 1;
console.log(a);
console.log(b);
console.log(b);
```

expect:

```typescript
const a = 1;
console.log(a);
const b = a + 1;
console.log(b);
console.log(b);
```

Versions:

```
Version: 1.75.1 (Universal)
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:34:59.000Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin x64 22.3.0
Sandboxed: No
abracadabra: latest
```

**Comment by nicoespeon** [comment]
Indeed, there is an issue when computing the position for the extraction. It's probably looking for the highest possible scope (I don't recall the details) and doesn't consider the position of the references it has.

We probably can refine this so it's created just before the highest reference, like illustrated in your `expect`.

Thanks for reporting @tjx666, always appreciated!

**Comment by nicoespeon** [comment]
That was a specific one. Hopefully, your test was really helpful to repro. It was a combination of:
- Replacing multiple occurrences
- Extracting an expression that refers to a variable
- Having the variable declared in the common ancestor of all extracted occurrences

By default, we were declaring the new variable at the top of that ancestor, resulting in the reported bug.

I fixed that so if there is a variable referenced that is declared in the scope, we make sure to declare the extracted variable _after_.

I captured that with a test and it's passing now 👍 

Thank you @tjx666 for the detailed report. That was really, really helpful ❤️
